src/audio/smart_amp/smart_amp_maxim_dsm.c

```diff
@@ -173,20 +211,20 @@ static void maxim_dsm_fb_proc(struct smart_amp_mod_struct_t *hspk,
 	if (*w_ptr >= DSM_FB_BUF_SZ) {
 		if (is_16bit) {
 			for (x = 0; x < DSM_FRM_SZ; x++) {
-			/* Buffer ordering for DSM : IVIV... -> VV... II...*/
-				v[x] = buf.buf16[4 * x + 1];
-				i[x] = buf.buf16[4 * x];
-				v[x + DSM_FRM_SZ] = buf.buf16[4 * x + 3];
-				i[x + DSM_FRM_SZ] = buf.buf16[4 * x + 2];
+			/* Buffer ordering for DSM : VIVI... -> VV... II...*/
+				v[x] = buf.buf16[4 * x];
+				i[x] = buf.buf16[4 * x + 1];
+				v[x + DSM_FRM_SZ] = buf.buf16[4 * x + 2];
+				i[x + DSM_FRM_SZ] = buf.buf16[4 * x + 3];
```

Comment:
These look like separate fixes and should be in a different patch to license and binary.

Reply:
OK. Shall split the patch. Thanks.